### PR TITLE
Added a 'must_pass_mask'

### PR DESCRIPTION
**Pull request by EmmaRenauld**
I was not able to differentiate to of my bundles with all the criteria already present. I added a 'must_pass_mask' (I am totally open to another name!).

So far we have
- endpoint mask or head + tail: where the streamlines must start and end
- gt_mask: the mask that will be used to compute the statistics (Overlap, Overreach)
- limits_mask: all streamlines must entirely stay inside that mask.

This adds:
- must_pass_mask: streamlines must pass through that region to be included in the bundle.

(Careful if you accept Antoine's PR before mine, I'll check if it needs refactoring)

@AntoineTheb

### CODE REVIEW

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

File: scilpy/tractanalysis/scoring.py [hunk 1]
```diff
@@ -396,6 +402,20 @@ def extract_vb_vs(
         wpc_ids.extend(out_of_mask_ids)
         vs_ids = np.setdiff1d(vs_ids, wpc_ids)
 
+    # Remove streamlines no passing through must_pass_mask
```
Comment (AntoineTheb) [inline scilpy/tractanalysis/scoring.py:405]: not passing

**Review by AntoineTheb — COMMENTED** [review]
Nice ! I would maybe suggest "intermediate" instead of "must pass" ?

**Comment by EmmaRenauld** [comment]
> Nice ! I would maybe suggest "intermediate" instead of "must pass" ?

@mdesco I added this to solve the CC issue. Now streamlines from the CC must pass trough the low mid-brain CC area. What would be the name you would use?

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

**Review by mdesco — APPROVED** [review]
All good to me. I have no strong opinion between "must_pass" or "intermediate". You guys decide @EmmaRenauld @AntoineTheb @arnaudbore

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-599/5/

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by EmmaRenauld** [comment]
Now that PR #614 is done, I merged, and tested again.

@arnaudbore, do you have an opinion on the mask name? Else, this is ready.

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by EmmaRenauld** [comment]
@arnaudbore asked me to change the names to ALL and ANY masks. This is done, and re-tested.
 This is ok with you, @AntoineTheb ?

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by AntoineTheb** [comment]
@EmmaRenauld good for me ! Do you have data to test this PR ?

**Comment by EmmaRenauld** [comment]
> @EmmaRenauld good for me ! Do you have data to test this PR ?

I tested it yesterday. It should be good.

**Review by arnaudbore — APPROVED** [review]

**Pull request merged**